service/storage_service.cc

```diff
@@ -1279,6 +1286,11 @@ class join_node_rpc_handshaker : public service::group0_handshaker {
     future<> pre_server_start(const group0_info& g0_info) override {
         rtlogger.info("join: sending the join request to {}", g0_info.ip_addr);
 
+        co_await utils::get_local_injector().inject("crash_before_group0_join", [](auto& handler) -> future<> {
+            // Sleep five seconds to ensure node gossips its state before crashing.
+            co_await sleep(std::chrono::seconds(5));
```

Comment:
Another reliance on timing... There **must** be a way to get rid of timing dependencies completely.

Reply:
Done  -- waiting for the node to finish sending its first `gossip_rpc_verbs::send_gossip_digest_ack2` message